File: ports/go/src/gui/widgets/bcsinput.go
package widgets

import (
	"sync"
	core "github.com/robertpelloni/btk-go/ports/go/src/core/kernel"
	gui "github.com/robertpelloni/btk-go/ports/go/src/gui/kernel"
)

type BcsInput struct {
	*gui.BcsWidget
	text string
	mu   sync.RWMutex
}

func NewBcsInput(parent *gui.BcsWidget) *BcsInput {
	var p *core.BcsObject
	if parent != nil {
		p = parent.BcsObject
	}
	i := &BcsInput{
		BcsWidget: gui.NewBcsWidget(p),
		text:      "",
	}
	i.Impl = i
	return i
}

func (i *BcsInput) SetText(t string) {
	i.mu.Lock()
	defer i.mu.Unlock()
	i.text = t
}

func (i *BcsInput) Text() string {
	i.mu.RLock()
	defer i.mu.RUnlock()
	return i.text
}

File: ports/java/src/gui/widgets/BcsInput.java
package gui.widgets;

import java.util.concurrent.locks.ReentrantLock;

public class BcsInput extends BcsWidget {
    private String text = "";
    private final ReentrantLock lock = new ReentrantLock();

    public BcsInput(BcsWidget parent) {
        super(parent);
    }

    public String getText() {
        lock.lock();
        try {
            return text;
        } finally {
            lock.unlock();
        }
    }

    public void setText(String text) {
        lock.lock();
        try {
            this.text = text;
        } finally {
            lock.unlock();
        }
    }
}
